hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/DispatchHandleContext.java

```diff
@@ -98,11 +101,15 @@ public class DispatchHandleContext implements HandleContext, FeeContext {
     private final DispatchMetadata dispatchMetaData;
     private final TransactionChecker transactionChecker;
     private final TransactionCategory transactionCategory;
+
     // This is used to store the pre-handle results for the inner transactions
     // in an atomic batch, null otherwise
     @Nullable
     private final List<PreHandleResult> preHandleResults;
 
+    @NonNull
+    private final BatchInnerTxnPreHandle innerTxnPreHandler;
```

Comment:
Instead of creating another type for people to grasp, I would lean to passing the `PreHandleWorkflow` singleton into each context and then using it when we need for a `BATCH_INNER`.